src/algebraic_geometry/height_function.lean

```diff
@@ -0,0 +1,392 @@
+/-
+Copyright (c) 2021 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import data.real.basic
+import algebra.category.Group.abelian
+import group_theory.finiteness
+import group_theory.coset
+import data.set.finite
+import algebra.geom_sum
+import analysis.specific_limits
+
+/-!
+# definition of height function and descent theorem (about finite-generatedness of abelian group)
+-/
+
+noncomputable theory
+
+open_locale pointwise big_operators
+open finset filter
+
+universe u
+
+variable (A : Ab.{u})
+
+/--definition of height function-/
```

Comment:
Please write a more detailed docstring, with capital letters and interpunction. Maybe add a reference to another source.